### PR TITLE
ChatEvent.isProxyCommand()

### PR DESCRIPTION
**Pull request by SimonOrJ**
This command returns `true` if the command will be intercepted by and run on BungeeCord.  The added code snippet is taken from [PluginManager.dispatchCommand()](https://github.com/SpigotMC/BungeeCord/blob/a47b8033857973588a0da98d3ebdc5d970176fa9/api/src/main/java/net/md_5/bungee/api/plugin/PluginManager.java#L129).

I was looking for this kind of command for easy command logging for some time now.  I hope this comes in as useful for some other developers out there.

### CODE REVIEW

**Comment by md-5** [comment]
Just duplicating code into an event is not a good fix for anything

**Comment by SimonOrJ** [comment]
The main thing I am trying to achieve is make an event method that can tell if the command is going to be run on the proxy server instead of being passed down to the Minecraft server.  I observed what the proxy server does to run the command on itself, then copied/coded the parts I believe is necessary for the method achieve its function.

One reason why I didn't modify `PluginManager.java` is because there is a better way to find out if the command can be run on the server within it.  The map variable with string-command pair `commandMap` does better than turning the variable into collections of entries then comparing each entry against the command name string.  I don't know if allowing access to commandMap would be ideal, so I left `PluginManager.java` as-is.

**Comment by SimonOrJ** [comment]
I've modified the code to leave the command checking logic in the `PluginManager.java` now

File: api/src/main/java/net/md_5/bungee/api/event/ChatEvent.java [hunk 1]
```diff
@@ -39,4 +41,16 @@ public boolean isCommand()
...
     }
+
+    /**
+     * Checks whether this message is run on this proxy server
+     *
+     * @return if this command runs on the proxy
+     */
+    public boolean isProxyCommand()
+    {
+        String commandName = message.split(" ",2)[0].substring( 1 );
```
Comment (Janmm14) [inline api/src/main/java/net/md_5/bungee/api/event/ChatEvent.java]: indexOf(" ") -> substring

Comment (SimonOrJ) [reply api/src/main/java/net/md_5/bungee/api/event/ChatEvent.java]: Great idea! I've never thought of using that faster method!

File: api/src/main/java/net/md_5/bungee/api/event/ChatEvent.java [hunk 2]
```diff
@@ -39,4 +41,16 @@ public boolean isCommand()
...
     }
+
+    /**
+     * Checks whether this message is run on this proxy server
+     *
+     * @return if this command runs on the proxy
+     */
+    public boolean isProxyCommand()
+    {
+        String commandName = message.substring( 1, message.indexOf( " " ) );
```
Comment (Janmm14) [inline api/src/main/java/net/md_5/bungee/api/event/ChatEvent.java]: If you use indexOf you need to handle it returning -1 aka missing. (Thought it was clear)

Comment (SimonOrJ) [reply api/src/main/java/net/md_5/bungee/api/event/ChatEvent.java]: oh right, that flew over my head

**Comment by Janmm14** [comment]
isProxyCommand should check isCommand beforehand

**Comment by SimonOrJ** [comment]
I should be the one remembering and noticing those at the first place.  I apologize for the missed code snippets, and thank you for doing the code reviews!

**Comment by md-5** [comment]
Thanks

**Pull request closed without merging**